DeleteRecordsAuthzIT sometimes fails with a sporadic NOT_LEADER_OR_FOLLOWER

**Describe the bug**

Following on from #3103 

Integration test `DeleteRecordsAuthzIT` sometimes fails with a sporadic NOT_LEADER_OR_FOLLOWER

```
mvn verify -Dit.test='DeleteRecordsAuthzIT' -DskipUTs  -Dfailsafe.failIfNoSpecifiedTests=false -pl kroxylicious-integration-tests -am
```

sometimes fails with a: 

```
[ERROR]   DeleteRecordsAuthzIT.shouldEnforceAccessToTopics:191->AuthzIT.verifyApiEqivalence:599 [Expect proxied response to be the same as the reference response with equivalent AuthZ] JSON documents are different:
Different value found in node "alice", expected: <"{
  "throttleTimeMs" : 0,
  "topics" : [ {
    "name" : "alice-topic",
    "partitions" : [ {
      "partitionIndex" : 0,
      "lowWatermark" : 0,
      "errorCode" : 0
    } ]
  } ]
}"> but was: <"{
  "throttleTimeMs" : 0,
  "topics" : [ {
    "name" : "alice-topic",
    "partitions" : [ {
      "partitionIndex" : 0,
      "lowWatermark" : -1,
      "errorCode" : 6
    } ]
  } ]
}">.
```

A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Running the above command line a loop
2. Run command '....'
3. See error

**Expected behavior**
Tests to pass

**YAML files and logs**

Attach or copy paste your kroxylicious configuration YAML (with sensitive details
like broker address redacted).
Attach or copy and paste relevant logs.

**Additional context**
Add any other context about the problem here.

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/authorization/AuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/authorization/AuthzIT.java
@@ -409,12 +409,7 @@ public abstract class AuthzIT extends BaseIT {
         }
         res.all().toCompletionStage().toCompletableFuture().join();
         Awaitility.waitAtMost(10, TimeUnit.SECONDS)
-                .until(() -> {
-                    Map<String, TopicDescription> join = admin.describeTopics(topicNames).allTopicNames().toCompletionStage().toCompletableFuture().join();
-                    return join.values().stream()
-                            .flatMap(topicDescription -> topicDescription.partitions().stream())
-                            .allMatch(p -> p.leader() != null);
-                });
+                .until(() -> allTopicPartitionsHaveALeader(admin, topicNames));
         return topicNames.stream().collect(Collectors.toMap(Function.identity(), topicName -> {
             try {
                 return res.topicId(topicName).get();
@@ -423,6 +418,13 @@ public abstract class AuthzIT extends BaseIT {
                 throw new RuntimeException(e);
             }
         }));
+    }
+
+    public static boolean allTopicPartitionsHaveALeader(Admin admin, List<String> topicNames) {
+        Map<String, TopicDescription> join = admin.describeTopics(topicNames).allTopicNames().toCompletionStage().toCompletableFuture().join();
+        return join.values().stream()
+                .flatMap(topicDescription -> topicDescription.partitions().stream())
+                .allMatch(p -> p.leader() != null);
     }
 
     protected static void ensureInternalTopicsExist(

